Formula/nimbus-eth2.rb

```diff
@@ -0,0 +1,49 @@
+class NimbusEth2 < Formula
+  desc "Nim implementation of the Ethereum 2.0 blockchain"
+  homepage "https://nimbus.guide/"
+  # pull from git tag to get submodules
+  url "https://github.com/status-im/nimbus-eth2.git",
+      tag:      "v1.5.5",
+      revision: "f0f9735955070d297b33fbec466537cb7029e0cc"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "make"
```

Comment:
This Makefile pulls in an entire Nim compiler toolchain, instead of using one provided by Homebrew. This should be fixed upstream before creating a formula.

Reply:
Can you clarify why is this considered a problem? I can imagine various motives:

1) All downloads of a recipe should be explicitly described through some homebrew-specific API, so they can be verified (against a checksum) and the recipe can be potentially executed in a network-isolated/offline environment.

2) All intermediate build files should be easy to clean up after the build and the Nim compiler dependency doesn't meet this criteria.

3) Building your compiler toolchain as part of your build is just considered a sub-optimal approach (i.e. an excessive waste of time).

I'd like to help in resolving the perceived issues, but first I need to understand your concerns better. I doubt Nimbus is the first project that builds a custom tool required only during the build process and I would argue that Nimbus doesn't depend on Nim at run-time, so we don't need to introduce a dependency on Nim in the Homebrew recipe.

Reply:
@zah Thank you for taking a look at this. The acceptance criteria are here: https://docs.brew.sh/Acceptable-Formulae

Specifically

> We don’t like install scripts that download unversioned things

If I understand correctly, the `nimbus_build_system` is built from `master` via a Git submodule.

A Homebrew maintainer is more qualified to comment on this

Reply:
Through the use of git submodules, we precisely pin the version of the Nim compiler being used as part of our build. We take reproducible builds very seriously, so this applies to all other components as well.

Reply:
@SMillerDev would you mind taking a look at this? (context: pulling in the compiler via Git submodule)

Reply:
That sounds like it's not exactly in line with this: https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

Reply:
Clearly, the security of Nimbus (a software dealing with finances in the form of cryptocurrencies) is of paramount importance. We, the team behind Nimbus, care very strongly about keeping our dependencies up-to-date with regard to security vulnerabilities and we pin the version of every component used during the build exactly in order to control precisely what goes into the shipped binaries. Here is a relevant except from our docs:

> We've designed the build process to be reproducible. In practice, this means that anyone can verify that these exact binaries were produced from the corresponding source code commits. For more about the philosophy and importance of this feature see [reproducible-builds.org](https://reproducible-builds.org/).

Reply:
I don't think that warrants an exception to our rules.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap